src/test/java/jewellery/inventory/unit/service/ProductServiceTest.java

```diff
@@ -0,0 +1,264 @@
+package jewellery.inventory.unit.service;
+
+import static jewellery.inventory.helper.ProductTestHelper.*;
+
+import jewellery.inventory.dto.request.ProductRequestDto;
+import jewellery.inventory.dto.request.resource.ResourceQuantityRequestDto;
+import jewellery.inventory.dto.response.ProductResponseDto;
+import jewellery.inventory.exception.not_found.*;
+import jewellery.inventory.exception.product.ProductIsContentException;
+import jewellery.inventory.exception.product.ProductIsSoldException;
+import jewellery.inventory.helper.ResourceTestHelper;
+import jewellery.inventory.helper.UserTestHelper;
+import jewellery.inventory.mapper.ProductMapper;
+import jewellery.inventory.mapper.UserMapper;
+import jewellery.inventory.model.Product;
+import jewellery.inventory.model.ResourceInUser;
+import jewellery.inventory.model.User;
+import jewellery.inventory.model.resource.Resource;
+import jewellery.inventory.repository.*;
+import jewellery.inventory.service.ProductService;
+import jewellery.inventory.service.ResourceInUserService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import java.util.*;
+
+@ExtendWith(MockitoExtension.class)
+class ProductServiceTest {
+
+  @InjectMocks private ProductService productService;
+  @Mock private ProductRepository productRepository;
+  @Mock UserMapper userMapper;
+  @Mock ProductMapper productMapper;
+  @Mock UserRepository userRepository;
+  @Mock ResourceRepository resourceRepository;
+  @Mock ResourceInUserRepository resourceInUserRepository;
+  @Mock ResourceInProductRepository resourceInProductRepository;
+  @Mock ResourceInUserService resourceInUserService;
+
+  private User user;
+  private Product testProduct;
+  private Resource pearl;
+  private ResourceInUser resourceInUser;
+  private ProductRequestDto productRequestDto;
+  private ResourceQuantityRequestDto resourceQuantityRequestDto;
+
+  @BeforeEach
+  void setUp() {
+
+    user = UserTestHelper.createTestUserWithRandomId();
+    pearl = ResourceTestHelper.getPearl();
+    resourceInUser = createResourceInUser(user, pearl);
+    testProduct = createTestProduct(user, pearl);
+    resourceQuantityRequestDto = createResourceQuantityRequestDto(pearl);
+    productRequestDto = createProductRequestDto(user, resourceQuantityRequestDto);
+  }
+
+  @Test
+  void testDeleteProductContainsProductBreakDown() {
+    testProduct.setProductsContent(List.of(new Product(), new Product(), new Product()));
+    when(productRepository.findById(testProduct.getId())).thenReturn(Optional.of(testProduct));
+
+    productService.deleteProduct(testProduct.getId());
+
+    assertEquals(0, testProduct.getProductsContent().size());
```

Comment:
You could also set contentOf of these products to be the testProduct. Then you can assert that after deletion they no longer have the contentOf fields set. This would be a more complete productBreakdown test.

Reply:
Done!